packages/purpose-outbound-writer/src/converters/toOutboundEventV2.ts

```diff
@@ -0,0 +1,94 @@
+import {
+  PurposeEventEnvelopeV2,
+  PurposeVersionV2,
+  PurposeV2,
+} from "pagopa-interop-models";
+import {
+  PurposeEvent as OutboundPurposeEvent,
+  PurposeVersionV2 as OutboundPurposeVersionV2,
+  PurposeV2 as OutboundPurposeV2,
+} from "@pagopa/interop-outbound-models";
+import { match } from "ts-pattern";
+import { Exact } from "../utils.js";
+
+function toOutboundPurposeVersionV2(
+  purposeVersion: PurposeVersionV2
+): Exact<OutboundPurposeVersionV2, PurposeVersionV2> {
+  return {
+    ...purposeVersion,
+    riskAnalysis: undefined,
+  };
+}
+
+function toOutboundPurposeV2(
+  purpose: PurposeV2
+): Exact<OutboundPurposeV2, PurposeV2> {
+  return {
+    ...purpose,
+    versions: purpose.versions.map(toOutboundPurposeVersionV2),
+    riskAnalysisForm: undefined,
+  };
+}
+
+export function toOutboundEventV2(
+  message: PurposeEventEnvelopeV2
+): OutboundPurposeEvent {
+  return match(message)
+    .returnType<OutboundPurposeEvent>()
+    .with(
+      { type: "PurposeAdded" },
+      { type: "DraftPurposeUpdated" },
+      { type: "PurposeWaitingForApproval" },
+      { type: "PurposeActivated" },
+      { type: "DraftPurposeDeleted" },
+      { type: "WaitingForApprovalPurposeDeleted" },
+      (msg) => ({
+        event_version: msg.event_version,
+        type: msg.type,
+        version: msg.version,
+        data: {
+          purpose: msg.data.purpose && toOutboundPurposeV2(msg.data.purpose),
```

Comment:
just curious, which is the difference with
```suggestion
          purpose: toOutboundPurposeV2(msg.data.purpose),
```
?

Reply:
the `purpose` property is optional on the `msg.data` object. `msg.data.purpose &&` check is needed so that `toOutboundPurposeV2` does not receive undefined